concordium_p2p_rpc.proto

```diff
@@ -4,223 +4,433 @@ import "google/protobuf/wrappers.proto";
 message NetworkChangeRequest {
+  // The identifier for the network.
   google.protobuf.Int32Value network_id = 1;
 }
 
+// A response containing information about the node.
 message NodeInfoResponse {
+  // The unique node identifier.
   google.protobuf.StringValue node_id = 1;
+
+  // The local time of the node represented as a unix timestamp in seconds.
   uint64 current_localtime = 2 [jstype = JS_STRING];
+
+  // The node type. Either "Node" or "Bootstrapper".
   string peer_type = 3;
+
+  // Whether the node is a baker.
   bool consensus_baker_running = 4;
+
+  // Whether consensus is running.
+  // This is only false if the protocol was updated to a version which the node software does not support.
   bool consensus_running = 5;
+
+  // Whether the node is "Active" or "Passive".
+  // - Active means that it is either a baker or a member of the finalization committee.
+  // - Passive means that it is neither a baker or a member of the finalization committee.
   string consensus_type = 6;
+
   enum IsInBakingCommittee {
+    // The node is not the baking committee.
     NOT_IN_COMMITTEE = 0;
+    // The node has valid baker keys that will be included in the baker committee in the next chain epoch.
     ADDED_BUT_NOT_ACTIVE_IN_COMMITTEE = 1;
+    // The node has invalid baker keys.
     ADDED_BUT_WRONG_KEYS = 2;
+    // The node has valid baker keys and is active in the baker committee.
     ACTIVE_IN_COMMITTEE = 3;
   }
+
+  // The baking status of the node.
   IsInBakingCommittee consensus_baker_committee = 7;
+
+  // Whether the node is part of the finalizer committee.
   bool consensus_finalizer_committee = 8;
+
+  // The baker id. This will be `null` if the node is not a baker.
   google.protobuf.UInt64Value consensus_baker_id = 9;
+
+  // Deprecated.
   google.protobuf.StringValue staging_net_username = 50 [deprecated = true];
 }
 
+// Hash of a block (encoded in hex). Is always 64 characters long.
+// Example: "987d6c06256fbf874d6ba14f19baee4390a31c6ee58edd9cc4efef62e89d22d7"
 message BlockHash {
   string block_hash = 1;
 }
 
+// An account address. Uses a base58-check encoding with a version byte set to 1. Is always 50 characters long.
+// Example: "3DJoe7aUwMwVmdFdRU2QsnJfsBbCmQu1QHvEg7YtWFZWmsoBXe"
 message AccountAddress {
   string account_address = 1;
 }
 
+// Hash of a transaction (encoded in hex). Is always 64 characters long.
+// Example: "987d6c06256fbf874d6ba14f19baee4390a31c6ee58edd9cc4efef62e89d22d7"
 message TransactionHash {
   string transaction_hash = 1;
 }
 
+// Request for getting the ancestors of a block.
 message BlockHashAndAmount {
+  // The block to get ancestors of.
   string block_hash = 1;
+
+  // The maximum amount of ancestors that will be returned.
   uint64 amount = 2 [jstype = JS_STRING];
 }
 
+// Reqest for sending a transaction.
 message SendTransactionRequest {
+  // The network id. Default is 100.
   uint32 network_id = 1;
+
+  // The transaction payload in binary encoding.
```

Comment:
Perhaps include a link to the developer website where this is outlined for some transactions.